plugins/kubernetes/test/models/kubernetes/template_filler_test.rb

```diff
@@ -187,9 +181,25 @@
       end
 
       it "fails when vault is not configured" do
-        Samson::Secrets::VaultClient.client.expects(:client).returns(nil)
-        e = assert_raises { template.to_hash }
-        e.message.must_equal "Could not find Vault config for pod1"
+        with_env('SECRET_STORAGE_BACKEND': "SecretStorage::HashicorpVault") do
+          Samson::Secrets::VaultClient.client.expects(:client).raises("Could not find Vault config for pod1")
+          e = assert_raises { template.to_hash }
+          e.message.must_equal "Could not find Vault config for pod1"
+        end
+      end
+
+      it "get the VAULT_ADDR in it's env" do
+        with_env('SECRET_STORAGE_BACKEND': "SecretStorage::HashicorpVault") do
+          assert template.to_hash[:spec][:template][:spec][:containers].
+            first[:env].any? { |env| env.any? { |_k, v| v == "VAULT_ADDR" } }
+        end
+      end
+
+      it "does not get the VAULT_ADDR in it's env" do
+        with_env('SECRET_STORAGE_BACKEND': "foobar") do
+          refute template.to_hash[:spec][:template][:spec][:containers].
```

Comment:
reused logic ... extract into a let ... otherwise could lead to false positives when someone only updates one of them